File: shanny-blog-java/blog-pojo/src/main/java/com/shanny/entity/Article.java
package com.shanny.entity;

import com.shanny.converter.LongListConverter;
import com.shanny.enums.ArticleEnum;
import com.shanny.enums.CategoryEnum;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
@Table(name = "articles")
public class Article {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Column(length = 50, nullable = false)
    private String title;

    @NotNull
    @Lob
    @Column(columnDefinition = "TEXT", nullable = false)
    private String content;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String memo;

    @Column(length = 300)
    private String image;

    @Column(length = 300)
    private String href;

    @Convert(converter = LongListConverter.class)
    private List<Long> tags;

    @NotNull
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private CategoryEnum.CategoryType type; // 内容类型

    @Column()
    private Long categoryId; // 内容分类

    @NotNull
    @Column(nullable = false)
    @Convert(converter = LongListConverter.class)
    private List<Long> timelines;

    @NotNull
    @Column(nullable = false)
    private Integer views = 0;

    @NotNull
    @Column(nullable = false)
    private Boolean published = true;

    @NotNull
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @Column(nullable = false)
    private LocalDateTime createTime;

    @NotNull
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @Column(nullable = false)
    private LocalDateTime updateTime;
}

File: shanny-blog-net/blog-pojo/Entities/Article.cs
using blog_common.Enums;
using blog_pojo.Converters;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace blog_pojo.Entities
{
    [Table("articles")]
    public class Article
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id")]
        public long Id { get; set; }

        [MaxLength(50)]
        [Required]
        [Column("title")]
        public string Title { get; set; } = string.Empty;

        [Column("content", TypeName = "TEXT")]
        [Required]
        public string Content { get; set; } = string.Empty;

        [Column("memo", TypeName = "TEXT")]
        [Required]
        public string Memo { get; set; } = string.Empty;

        [MaxLength(300)]
        [Column("image")]
        public string Image { get; set; } = string.Empty;

        [MaxLength(300)]
        [Column("href")]
        public string Href { get; set; } = string.Empty;

        [Column("tags")]
        public List<long> Tags { get; set; } = new();

        [Required]
        [Column("type")]
        public CategoryType Type { get; set; }

        [Column("categoryId")]
        public long? CategoryId { get; set; }

        [Required]
        [Column("timelines")]
        public List<long> Timelines { get; set; } = new();

        [Required]
        [Column("views")]
        public int Views { get; set; } = 0;

        [Required]
        [Column("published")]
        public bool Published { get; set; } = true;

        [Required]
        [Column("createTime")]
        public DateTime CreateTime { get; set; }

        [Required]
        [Column("updateTime")]
        public DateTime UpdateTime { get; set; }
    }
}
